### ISSUE TITLE
IDE0027 false positive (Simplify collection initialization)

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#16158]
**Version Used**: Visual Studio 2017 RC 15.0.26014.0

**Code to Reproduce**:
```csharp
string item = null;
var items = new List<string>();

var values = new List<string>(); // Collection initialization can be simplified
values.Add(item);
values.AddRange(items);
```

**Actual Behavior**:
```csharp
string item = null;
var items = new List<string>();

var values = new List<string>
{
    item,
    items // error
};
```
